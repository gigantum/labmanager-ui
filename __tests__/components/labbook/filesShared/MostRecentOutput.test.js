
      import React, {Component} from 'react'
      import renderer from 'react-test-renderer';
      import {mount} from 'enzyme'
      import MostRecentOutput from 'Components/labbook/filesShared/MostRecentOutput';
      import store from 'JS/redux/store'

      import {BrowserRouter as Router, Switch, Route} from 'react-router-dom'

      import json from './__relaydata__/MostRecentOutput.json'

      import relayTestingUtils from 'relay-testing-utils'

      const fixtures = {
        edgeId: json.data.labbook.output.id,
        output: json.data.labbook.output
      }

      store.dispatch({
        type: 'UPDATE_CALLBACK_ROUTE',
        payload: {
          callbackRoute: '/labbooks/username/labbookName/outputData'
        }
      })

      class MostRecentOutputComponent extends Component{
        render(){
          return(relayTestingUtils.relayWrap(<MostRecentOutput {...fixtures}/>, {}, json.data.labbook.output))
        }
      }

      describe('Test MostRecentOutput', () => {
        it('render snapshot', ()=>{
          const wrapper = renderer.create(
             <Router>
               <Switch>

                 <Route

                   path=""
                   render={(props) =>
                     <MostRecentOutputComponent />
                   }
                 />

               </Switch>
             </Router>
          );

          const tree = wrapper.toJSON()

          expect(tree).toMatchSnapshot()
        })


      })
