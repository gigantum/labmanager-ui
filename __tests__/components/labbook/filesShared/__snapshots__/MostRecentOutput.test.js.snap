// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MostRecentOutput render snapshot 1`] = `
<div
  className="FileEmpty"
>
  <div
    className="FileEmpty__container FileEmpty__container--outputData"
  >
    <p
      className="FileEmpty__main-text"
    >
      This Project has No Recent Files
    </p>
    <a
      className="FileEmpty__sub-text"
      href="../../../../projects///outputData"
      onClick={[Function]}
    />
  </div>
</div>
`;
